src/data/polynomial/derivative.lean

```diff
@@ -204,6 +204,10 @@ begin
   exact hf h2
 end
 
+theorem eq_C_of_derivative_eq_zero [no_zero_divisors R] [char_zero R] {f : R[X]}
```

Comment:
If you want this can be (slightly) generalized to `[no_zero_smul_divisors ℕ R]` instead of `[char_zero R] [no_zero_divisors R]`. Here is a proof of `nat_degree_eq_zero_of_derivative_eq_zero`.
```lean
theorem nat_degree_eq_zero_of_derivative_eq_zero  [no_zero_smul_divisors ℕ R] {f : R[X]}
  (h : f.derivative = 0) : f.nat_degree = 0 :=
begin
  rcases eq_or_ne f 0 with rfl | hf,
  { exact nat_degree_zero },
  rw nat_degree_eq_zero_iff_degree_le_zero,
  by_contra' f_nat_degree_pos,
  rw [←nat_degree_pos_iff_degree_pos] at f_nat_degree_pos,
  let m := f.nat_degree - 1,
  have hm : m + 1 = f.nat_degree := tsub_add_cancel_of_le f_nat_degree_pos,
  have h2 := (coeff_derivative f m).symm,
  rw polynomial.ext_iff at h,
  rw [h m, coeff_zero, show (↑m : R) + 1 = ↑(m + 1), by simp, ← nsmul_eq_mul', smul_eq_zero] at h2,
  replace h2 := h2.resolve_left (λ h2, add_one_ne_zero _ h2),
  rw [hm, ←polynomial.leading_coeff, polynomial.leading_coeff_eq_zero] at h2,
  exact hf h2
end
```